src/usePromise.ts

```diff
@@ -22,7 +28,7 @@ export default function usePromise<T>(
     if (!promiseOrFn) {
       setState({
         loading: false,
-        error: null,
+        error: new TypeError(`The argument passed to usePromise must be either a promise or a function, but you passed a ${typeof promiseOrFn}`),
```

Comment:
Alternatively, this could throw a type error instead b/c a falsy value in `promiseOrFn` doesn't satisfy the type specified for the hook argument. Throwing an error might be better than setting an error on the state since this has the potential to be incorrectly interpreted as a rejected promise. 🤷 Would like your input on this please.